cmd/templ/generatecmd/eventhandler_test.go

```diff
@@ -0,0 +1,61 @@
+package generatecmd
+
+import (
+	"context"
+	"go/scanner"
+	"go/token"
+	"io"
+	"log/slog"
+	"testing"
+
+	"github.com/a-h/templ/generator"
+)
+
+func TestEventHandler(t *testing.T) {
+	tests := []struct {
+		name           string
+		fileName       string
+		errorPositions []token.Position
+	}{
+		{
+			name:     "single error outputs location in srcFile",
+			fileName: "single_error.templ.error",
```

Comment:
The `.error` suffix was necessary to prevent the test files from breaking the build because they would get pulled into any `templ generate` running from the root.